build-tools-internal/src/integTest/groovy/org/elasticsearch/gradle/internal/ElasticsearchJavaModulePathPluginFuncTest.groovy

```diff
@@ -23,6 +25,19 @@ class ElasticsearchJavaModulePathPluginFuncTest extends AbstractJavaGradleFuncTe
 
     public static final String ES_VERSION = VersionProperties.getElasticsearch()
 
+    public static final String COMPILE_JAVA_CONFIG = """
+        tasks.named('compileJava').configure {
+            doLast {
+                def sep = org.gradle.internal.os.OperatingSystem.current().isWindows() ? ':' : ';'
```

Comment:
We can switch to `org.elasticsearch.gradle.OS` here and avoid using gradle internal api